core/src/com/jediterm/terminal/model/JediTerminal.java

```diff
@@ -1090,7 +1090,7 @@ private void adjustXY(int dirX) {
           myCursorX += 1;
         }
       } else {
-        myCursorX -= 1; //low surrogate character can't be the first character in the line
+        if (myCursorX > 0) myCursorX -= 1; //low surrogate character can't be the first character in the line
```

Comment:
Maybe the more consistent coding style would be `myCursorX = Math.max(0, myCursorX - 1);`